### PR TITLE
Add ignore lock flag

### PR DESCRIPTION
**Pull request by gmazzap**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?**

A new `--ignore-lock` flag is added to the `compile-assets` command.

* **Does this PR introduce a breaking change?**

No.

* **Other information**:

The way the new flag work is documented here: https://github.com/inpsyde/composer-asset-compiler/blob/add-ignore-lock-flag/README.md#ignoring-lock

### CODE REVIEW

File: src/Composer/Command/CompileAssets.php [hunk 1]
```diff
@@ -73,10 +80,20 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
             $plugin->activate($composer, $io);
 
-            $noDev = (bool)$input->hasOption('no-dev');
+            $noDev = $input->hasOption('no-dev');
             $env = $input->hasOption('env') ? $input->getOption('env') : null;
 
-            $plugin->runByCommand(is_string($env) ? $env : null, !$noDev);
+            $ignoreLock = '';
+            if ($input->hasOption('ignore-lock')) {
+                $ignored = $input->getOption('ignore-lock');
```
Comment (cristianobaptista) [inline src/Composer/Command/CompileAssets.php:88]: @gmazzap The call to `$input->hasOption('ignore-lock')` is always returning true even when I don't set the `--ignore-lock` option and `$input->getOption('ignore-lock')` always return an empty string if I don't pass a value, so just adding the option without a value will not fallback to setting the value to `*` and no asset compilation will be forced.

**Review by cristianobaptista — CHANGES_REQUESTED** [review]
@gmazzap I've tested setting the `--ignore-lock` value to `*`, single packages and comma-separated lists of target packages and namespaces with wildcard and everything worked perfectly.

There is just the issue with setting `--ignore-lock` without parameters, not really working for me.

File: src/Composer/Command/CompileAssets.php (hunk 1 above)
Comment (gmazzap) [reply src/Composer/Command/CompileAssets.php]: Thanks @cristianobaptista

According to Symfony documentation, `$input->getOption` is supposed to to return the default value when flag is provided without value, and return `null` when the flag is not there at all... but for some reason it's not working like that, maybe it's Composer that changes the behavior.

Switching to the low-level `hasParameterOption` should have solved the issue now.

Could you please test again?

Comment (cristianobaptista) [reply src/Composer/Command/CompileAssets.php]: @gmazzap All good now after performing once again all tests, thanks!

**Comment by cristianobaptista** [comment]
@gmazzap after running again my tests, I realized that it might be a good idea to optionally ignore pre-compiled assets or maybe even make it the default because if there are pre-compiled assets, those will be used and no local code changes will be taken into account. What do you think?

**Comment by gmazzap** [comment]
@cristianobaptista unlike thr lock file, pre-compiled assets are opt-in: you need to have specific configuration to make it work.

If no configuration is there, no pre-compiled asset will be used.

When some configuration will be there... it would/should be env-specific, something like this:

```json
{
    "pre-compiled": {
        "env": {
            "prod": {
                "source": "assets-${version}",
                "target": "./assets/",
                "adapter": "gh-release-zip"
            },
            "dev": {
                "source": "assets-${hash}",
                "target": "./assets/",
                "adapter": "gh-action-artifact"
            }
        }
    }
}
```

With such configuration, unless `--env=prod` or `--env=dev` is passed as argument 
 to the `composer compile-assets` command (or the `COMPOSER_ASSET_COMPILER` environment variable<sup>1</sup> is set to one of those two values), pre-compiled assets would be ignored.

This is why, to be honest, I wouldn't mix the `--ignore-lock` with pre-compiled assets (and in that case the flag should me renamed to something like `--recompile`).

Even because, in that case using both `--ignore-lock --env=prod` would be unclear which one takes precedence.

-----

<sup>1</sup> Personally, on my dev machine, I've  `COMPOSER_ASSET_COMPILER` environment variable globally set to **`local`**, to have that default value when no `--env` flag is used.

**Comment by cristianobaptista** [comment]
@gmazzap sounds reasonable to me. Thanks for your clarification and insights 😁

**Review by cristianobaptista — APPROVED** [review]
@gmazzap all good with this PR from my perspective :+1:

**Pull request merged**